File: src/scripting/examples/example_remove_impair_lines.rb
#!/usr/bin/env ruby

# SPDX-FileCopyrightText: 2007-2009 Sergio Pistone <sergio_pistone@yahoo.com.ar>
#
# SPDX-License-Identifier: GPL-2.0-or-later

require "subtitle"

subtitle = Subtitle.instance()
for line_index in (0..subtitle.linesCount()-1).to_a().reverse()
	subtitle.removeLine( line_index ) if ( line_index % 2 == 0 )
end

File: src/scripting/examples/example_remove_impair_lines.js
/*
    SPDX-FileCopyrightText: 2007-2009 Sergio Pistone (sergio_pistone@yahoo.com.ar)

    SPDX-License-Identifier: GPL-2.0-or-later
*/

s = subtitle.instance();
for ( var lineIndex = s.linesCount() - 1; lineIndex >= 0; --lineIndex )
{
	if ( lineIndex % 2 == 0 )
		s.removeLine( lineIndex );
}
